### ISSUE TITLE
selectizeInput option allowEmptyOption = TRUE is ignored (update selectize.js >= 0.12.2 needed)

### ISSUE THREAD

**Issue by mnazarov** [rstudio/shiny#1837]
I discovered that selectizeInput's option `allowEmptyOption = TRUE` doesn't work properly.
This is a bug in the selectize.js library version 0.12.1, as described in https://github.com/selectize/selectize.js/issues/739 - and was fixed in the version 0.12.2, so I propose to update the library.

A minimal example to see the issue:
```
shinyApp(
    ui = selectizeInput("letters", "Letters:", 
        choices  = c("(none)" = "", letters[1:5]), 
        multiple = FALSE, 
        options  = list(allowEmptyOption = TRUE)),
    server = function(input, output) {}
)
```

**Comment by jakelever** [comment]
+1

**Comment by phlogisticfugu** [comment]
making a fix for this, as I needed the updated version too

**Comment by alandipert** [comment]
Thank you, this was fixed by #2028
